packages/interledger/src/accounts/service.ts

```diff
@@ -790,122 +832,101 @@ export class AccountsService implements AccountsServiceInterface {
         return TransferError.UnknownSourceAccount
       } else {
         return TransferError.UnknownDestinationAccount
       }
     }
+    const sourceAccount =
+      accounts[0].id === sourceAccountId ? accounts[0] : accounts[1]
+    const destinationAccount =
+      accounts[0].id === destinationAccountId ? accounts[0] : accounts[1]
 
-    const transfers: ClientTransfer[] = []
-
-    const flags = 0 | TransferFlags.two_phase_commit
-    const timeout = BigInt(1e9)
+    const transfers: TwoPhaseTransfer[] = []
 
     if (sourceAccount.assetCode === destinationAccount.assetCode) {
       if (destinationAmount && sourceAmount !== destinationAmount) {
         return TransferError.InvalidDestinationAmount
       }
       transfers.push({
         id: randomId(),
-        debit_account_id: sourceAccount.balanceId,
-        credit_account_id: destinationAccount.balanceId,
+        sourceBalanceId: sourceAccount.balanceId,
+        destinationBalanceId: destinationAccount.balanceId,
         amount: sourceAmount,
-        flags,
-        timeout,
-        reserved: TRANSFER_RESERVED,
-        code: 0,
-        user_data: BigInt(0),
-        timestamp: BigInt(0)
+        twoPhaseCommit: true
       })
     } else {
       if (!destinationAmount) {
         return TransferError.InvalidDestinationAmount
       }
-
       transfers.push(
         {
           id: randomId(),
-          debit_account_id: sourceAccount.balanceId,
-          credit_account_id: toLiquidityId({
+          sourceBalanceId: sourceAccount.balanceId,
+          destinationBalanceId: toLiquidityId({
             assetCode: sourceAccount.assetCode,
             assetScale: sourceAccount.assetScale,
             hmacSecret: this.config.hmacSecret
           }),
           amount: sourceAmount,
-          flags: flags | TransferFlags.linked,
-          timeout,
-          reserved: TRANSFER_RESERVED,
-          code: 0,
-          user_data: BigInt(0),
-          timestamp: BigInt(0)
+          twoPhaseCommit: true
         },
         {
           id: randomId(),
-          debit_account_id: toLiquidityId({
+          sourceBalanceId: toLiquidityId({
             assetCode: destinationAccount.assetCode,
             assetScale: destinationAccount.assetScale,
             hmacSecret: this.config.hmacSecret
           }),
-          credit_account_id: destinationAccount.balanceId,
+          destinationBalanceId: destinationAccount.balanceId,
           amount: destinationAmount,
-          flags,
-          timeout,
-          reserved: TRANSFER_RESERVED,
-          code: 0,
-          user_data: BigInt(0),
-          timestamp: BigInt(0)
+          twoPhaseCommit: true
         }
       )
     }
-    const res = await this.client.createTransfers(transfers)
-    for (const { index, code } of res) {
-      switch (code) {
+    const error = await this.createTransfers(transfers)
+    if (error) {
+      switch (error.code) {
```

Comment:
Nit: The `if { switch { ` blocks (here and elsewhere in this file) could be simplified to `switch (error?.code) {` if you add a noop `null`/`undefined` `case`.

Reply:
Huh, that's annoying. Would `error && error.code` work instead?

Reply:
It complains about the subsequent `error.index`s and `error.code`

Reply:
Ok, never mind then.